src/js/chart.js

```diff
@@ -0,0 +1,193 @@
+    let dataPoint = {
+      label: (i === 0) ? `Latest:${info.buildNum[i]}` : info.buildNum[i],
+      y: info.buildTime[i],
+      color: info.buildColor[i],
+      toolTipContent: info.buildInfo[i],
+      click: onClick,
+      cursor: 'pointer'
+    };
+    dataPoints.push(dataPoint);
+  }
+  return dataPoints;
+};
+
+const buildChart = (info, data) => {
+  return new CanvasJS.Chart('chartContainer', {
+    width: $('#chartHeader').width(),
+    height: 298,
+    backgroundColor: bodyBgColor,
+    animationEnabled: true,
+    title: {
+      text: 'Build Status (Recent 10 builds)',
+      fontFamily: GLOBAL_FONT_FAMILY,
+      fontWeight: 'normal',
+      fontColor: GITHUB_GREY,
+      fontSize: 20
+    },
+    axisX: {
+      title: 'Build Number',
+      titleFontFamily: GLOBAL_FONT_FAMILY,
+      titleFontWeight: 'normal',
+      titleFontSize: 12
+    },
+    axisY: {
+      title: 'Build Time (Minutes)',
+      titleFontFamily: GLOBAL_FONT_FAMILY,
+      titleFontWeight: 'normal',
+      titleFontSize: 12
+    },
+    toolTip:{
+      fontFamily: GLOBAL_FONT_FAMILY
+    },
+    data: [{
+        type: 'column', //change type to bar, line, area, pie, etc
+        dataPoints: assembleDataPoints(info, data)
+    }]
+  });
+};
+
+const trimBuildNums = (range, info) => {
+  for (let i = range; i < columnAmount; i++) {
+    if (typeof info.buildNum[i] === 'undefined') {
+      info.buildNum[i] = '#';
+    }
+  }
+};
+
+const mapBuildData = (range, data) => {
+  let buildNum = [],
+    buildTime = [],
+    buildColor = [],
+    buildInfo = [];
+  const GREEN_INDEX = 1;
+
+  for (let i = 0; i < range; i++) {
+    let buildDuration = trimTimeToMinScale(data[i]['duration']);
+    let buildState = data[i]['state'];
+    let buildResult = data[i]['result'];
+    let buildMessage = trimMessage(data[i]['message']);
+    let buildStarted = data[i]['started_at'];
+    let buildFinished = data[i]['finished_at'];
+    let buildTimeStr = trimTimeStr(data[i]['duration']);
+
+    buildNum.push(`#${data[i]['number']}`);
+    buildInfo.push(`<b>Build Time</b>: ${buildTimeStr}<br/><span><b>Message: </b>${buildMessage}</span>`);
+    buildTime.push(buildDuration);
+
+    if (buildState === 'started') {
+      buildColor.push(allColor[2]);
+      if (buildStarted && buildFinished === null) {
+        let skipTime = (new Date() - new Date(buildStarted)) / 1000;
+        let skipTimeStr = trimTimeStr(skipTime);
+        buildTime[i] = trimTimeToMinScale(skipTime);
+        buildInfo[i] = `It's running! <b>Skipped time</b>:${skipTimeStr}<br/><span><b>Message:</b>${buildMessage}</span>`;
+      } else {
+        buildInfo[i] = `Oops, the build may be cancelled.<br/><span><b>Message:</b>${buildMessage}</span>`;
+      }
+    } else {
+      buildResult = (buildResult === null) ? GREEN_INDEX : buildResult;
+      buildColor.push(allColor[buildResult]);
+    }
+  }
+  return {buildNum: buildNum, buildTime: buildTime, buildColor: buildColor, buildInfo: buildInfo};
+};
```

Comment:
This can be done simply with the enhanced object literal in ES6: 

```js
return {
    buildNum, buildTime, buildColor, buildInfo
}
```

Google with: [es6 enhanced object literals](https://www.google.com/search?q=es6+enhanced+object+literals)